numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -464,8 +472,23 @@ PyUFunc_SimpleUniformOperationTypeResolver(
         PyObject *type_tup,
         PyArray_Descr **out_dtypes)
 {
-    const char *ufunc_name = ufunc_get_name_cstr(ufunc);
 
+    static PyObject *visibleDeprecationWarning = NULL;
+    npy_cache_import(
+        "numpy", "VisibleDeprecationWarning",
+         &visibleDeprecationWarning);
+    if (visibleDeprecationWarning == NULL) {
+                return -1;
+    }
+    const char *ufunc_name = ufunc_get_name_cstr(ufunc);
+    if (PyArray_ISCOMPLEX(operands[0]) &&
```

Comment:
What about the other operand here?

Reply:
This is operation on a single ndarray, hence we do not need check more i assume.

Reply:
Actually, it's not. You can see the signature for `fmax`, `fmin`, `minimum` and `maximum` here: https://numpy.org/doc/stable/reference/ufuncs.html#comparison-functions

All of them also have `x2`. Though I do realize why this can be confusing: `np.max` and `np.min` only take a single array. However, `np.max` is (roughly) `np.maximum.reduce`, and you can find the definition for `ufunc.reduce` here: https://numpy.org/doc/stable/reference/generated/numpy.ufunc.reduce.html#numpy.ufunc.reduce

Reply:
Ah I understand

Reply:
The uniform type resolvers output are uniform types though. So you may be able to only check one of them if you move this to the end of the function.

In fact, I may suggest creating a `PyUFunc_SimpleUniformOperationTypeResolverDeprecateComplex` which just calls the original version and then does this check. I am not a big fan of the `strcmp` here, and its not necessary (you will have to change type resolver in the ufunc API creation python script in code generators.

Reply:
@seberg  Sure will look into that approach.